README.md

```diff
@@ -1,159 +1,136 @@
 
-First we install the libraries used by the project:
+### Environment variables run-time injection
 
-```
-$ yarn install
-```
+The frontend container needs to adapt to different environments, reading at
+run-time environment variables values. For example, the application needs to
+know the address of the backend application. This is a non trivial task, since
+Javascript code runs on the client machine of the user executing the
+application, which prevents the application from directly reading the
+environment variables from the container.
 
-### Generating translations and types
+To overcome this limitation, an *env.sh* bash script inside `scripts` folder is
+executed every time the frontend application container starts. The script reads
+the environment variables and produces an *env-config.js* file that is then
+automatically copied together with the rest of the files to be served by the
+webserver. The *index.html* file (in the *public* folder of this repository)
+links already to *env-config.js*, which is read every time the user opens the
+application in a browser.
 
-The second step is to generate the definitions from the YAML translations and from OpenApi specs.
+>**IMPORTANT**: The *env.sh* script reads and automatically injects in
+>`env-config.js` all environment variables prefixed with *IO_ONBOARDING_PA*, for
+>example *IO_ONBOARDING_PA_API_HOST*.
 
-To generate the definitions from the YAML translations:
+To read the variable values inside the frontend application, use
+`window._env_.IO_ONBOARDING_PA_YOUR_VAR`. 
 
-```
-$ yarn generate:locales
-```
+## Production deployment
 
-To generate the definitions from the OpenApi specs:
+The frontend docker image is deployed in production on top of a Kubernetes
+cluster.
 
-```
-$ yarn generate:locales
-```
+For more informations about IO production deployments and helm-charts, see
+[io-infrastructure-post-config
+repository](https://github.com/teamdigitale/io-infrastructure-post-config).
 
-These commands can be run together using:
+Currently an Azure development pipeline is configured in order to deploy the
+master branch of this repository into the IO infrastructure staging environment;
+the process is triggered by every push into master.
 
-```
-$ yarn generate:all
-```
+The deployed frontend is reachable at
+[https://pa-onboarding.dev.io.italia.it/](https://pa-onboarding.dev.io.italia.it/).
 
 ## Test the application locally
 
-The application can be built and tested locally, either directly on the developer machine (directly using development tools, such as *yarn* or *parcel*), or as a Docker-based image.
-
-### Test locally with mocked APIs
-
-To test the frontend application locally, on developer machine, against some mocked APIs, run in the root folder
+Set up your environment variables first:
 
 ```shell
-yarn start-with-mock
+cp env.example .env
+# edit .env file
 ```
```

Comment:
with  the copied env.example into .env the mocked server and the app should work without any changes?

Reply:
yes, it should